core/gui/src/app/dashboard/component/user/user-dataset/user-dataset-explorer/user-dataset-file-editor/user-dataset-file-editor.component.html

```diff
@@ -0,0 +1,230 @@
+<!--
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+-->
+
+<!-- Loading State -->
+<div *ngIf="isLoading" class="file-loading">
```

Comment:
Few high level comments regarding this component:
1. Can you merge it with the `file-renderer` component: if a file is editable, 
<img width="82" height="37" alt="Screenshot 2025-08-13 at 11 44 24 PM" src="https://github.com/user-attachments/assets/a5fb8afa-2582-42ab-990a-f613820b54ff" /> show an edit button. And when users click that button, show the editing interface. This can avoid the redundancy file header.

2. I still want you to explore the library way of having the markdown editor. Implementing one by ourselves brings high maintenance cost and can be error-pruning.

Reply:
This project is worth checking out: https://github.com/ghiscoding/angular-markdown-editor